### PR TITLE
EASYOPAC-719 - Extending todays openning hours with tabs by category.

### PR DESCRIPTION
**Pull request by bozeak**

### CODE REVIEW

**Comment by guddo** [comment]
@bozeak Can't this be made as a module instead of Core?

**Comment by bozeak** [comment]
@guddo Theoretically it is possible.

**Comment by guddo** [comment]
@bozeak But not practically? As I said in office, I don't like changes in core 😅

**Comment by bozeak** [comment]
@guddo And practically too, but will have to spend some time on moving of functionality into separate module and reusage of core module functionality.

File: modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.theme.inc [hunk 1]
```diff
@@ -273,3 +273,171 @@ function template_preprocess_ding_ddbasic_opening_hours_all(&$variables) {
     $structured
   );
 }
+
+/**
+ * Categorized library opening hours.
+ */
+function theme_ding_ddbasic_opening_hours_all_tabbed($variables) {
+  return render($variables['tabbed']);
```
Comment (NicolaeGrigoriu) [inline modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.theme.inc]: Based on the `template_preprocess_ding_ddbasic_opening_hours_all_tabbed` variable `$variables['tabbed']` is already rendered in line 442 (see https://github.com/easySuite/ding2/pull/344/files#diff-9cf518f5f8d70d2545eb3f003295b475R442). I suppose that only one rendering is necessary and one render function call can be removed.

File: modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.theme.inc [hunk 2]
```diff
@@ -273,3 +273,171 @@ function template_preprocess_ding_ddbasic_opening_hours_all(&$variables) {
...
+  }
+
+  // Get and sort all the instances in the given timespan.
+  $instances = opening_hours_instance_load_multiple(
+    array_values($order),
+    date('Y-m-d', $today),
+    date('Y-m-d', $today)
+  );
+
+  foreach ($order as $library_nid) {
```
Comment (NicolaeGrigoriu) [inline modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.theme.inc]: `$order` array is defined in the `nodequeue_load_nodes` foreach call. Is it possible that this($order) array to be empty? If so, I think it is better to define it somewhere before its usage.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 43156383c45618ea3ff16eb80dbfb666e239ae92
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/easySuite/ding2

---
This message was auto-generated by https://gitcop.com

**Comment by NicolaeGrigoriu** [comment]
There is issue with gitcop, it takes too much time to fix this. So, we agreed to merge as it is.

**Comment by guddo** [comment]
_**Hint**: `git rebase -i HEAD~2` and you can "reword" your commit._ /cc @NicolaeGrigoriu @bozeak

**Pull request merged**